### PR TITLE
Allow customizing tree

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
This PR adds `map` node type to OMERO.web (at the moment new type is hardcoded) and allows handling custom attributes passed to the tree. Additional bug fixes

To test:
 - this PR is tested with https://github.com/aleksandra-tarkowska/mapannotations
 - `pip install git+git://github.com/aleksandra-tarkowska/mapannotations.git`

 - and:

 ```
/home/omero/workspace/OMERO-web/OMERO.web/bin/omero config append omero.web.apps '"mapr"'
/home/omero/workspace/OMERO-web/OMERO.web/bin/omero config set omero.web.ui.top_links '[["IDR", {"viewname": "webindex", "query_string": {"experimenter": -1}}, {"title": "Image Data Repository"}], ["Genes", {"viewname": "maprindex_gene", "query_string": {"experimenter": -1}}, {"title": "Genes browser"}], ["Phenotypes", {"viewname": "maprindex_phenotype", "query_string": {"experimenter": -1}}, {"title": "Phenotypes browser"}], ["siRNA", {"viewname": "maprindex_sirna", "query_string": {"experimenter": -1}}, {"title": "siRNA browser"}], ["Compound", {"viewname": "maprindex_compound", "query_string": {"experimenter": -1}}, {"title": "Compound browser"}], ["History", "history", {"title": "History"}], ["About", "/about/"], ["Help", "http://help.openmicroscopy.org/", {"title":"Open OMERO user guide in a new tab", "target":"new"}]]'

 ```

for full example see console output in https://10.0.51.150:8443/job/OMERO-web/

*This PR could be discussed in terms of how pagination will be handled in the main data tree when mixed types of nodes are loaded per page. At the moment nodes without counter are not using pagination*

### CODE REVIEW

**Comment by joshmoore** [comment]
@aleksandra-tarkowska : is it time to configure this for m52?

**Comment by aleksandra-tarkowska** [comment]
would be good to get that deployed somewhere including https://github.com/aleksandra-tarkowska/mapannotations app

**Comment by joshmoore** [comment]
As with gh-4688, is this needed for the upcoming UI tests?

**Comment by aleksandra-tarkowska** [comment]
I am about to rebase this PR against develop, yes we want that PR in m4 (mapr needs that). #4688 is not related to this PR

**Comment by joshmoore** [comment]
`develop` or `dev_5_2`? `develop` will bring in a new DB (if you intend that for metadata52)

File: components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.right_panel_mapanns_pane.js [hunk 1]
```diff
@@ -59,66 +62,84 @@ var MapAnnsPane = function MapAnnsPane($element, opts) {
...
+                    $(".tooltip", $mapAnnContainer).tooltip_init();
                 }
-                html = html + mapAnnsTempl({'anns': client_map_annotations,
-                    'showTableHead': false, 'showNs': false, 'clientMapAnn': true});
-                html = html + mapAnnsTempl({'anns': map_annotations,
-                    'showTableHead': false, 'showNs': true, 'clientMapAnn': false});
-                $mapAnnContainer.html(html);
-
-                // Finish up...
-                OME.linkify_element($( "table.keyValueTable tbody tr td" ));
```
Comment (aleksandra-tarkowska) [inline components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.right_panel_mapanns_pane.js]: not in develop

File: components/tools/OmeroWeb/omeroweb/settings.py [hunk 2]
```diff
@@ -653,6 +653,19 @@ def leave_none_unset_int(s):
           " ] or [\"Repository\", {\"viewname\": \"webindex\", "
           "\"query_string\": {\"experimenter\": -1}}, "
           "{\"title\": \"Repo\"}]'``")],
+    "omero.web.ui.metadata_pane":
```
Comment (aleksandra-tarkowska) [inline components/tools/OmeroWeb/omeroweb/settings.py]: this allows to set custom order of metadata pane accordeon. Any suggestions about var naming, dict, etc. see https://github.com/openmicroscopy/openmicroscopy/pull/4704/commits/ea068154e8a62966af4ee4e388730d5b8f62fdf8#diff-5ecd969a9790526e02d15fcb673d6ca5R722 for how template use it

**Comment by joshmoore** [comment]
Note: last 4 commits will need to be backported to the mainline. cc: @jburel

**Comment by joshmoore** [comment]
@manics suggests stripping off those 4 commits as a separate PR and merging this proactively.

**Comment by joshmoore** [comment]
Note: this omits 49c396df733c7bab374ef0da1111606a4efd303f and 1b2ff38ca886fd604b0ff2946c4b7132ff9db630 since they don't affect the IDR. Merging.

**Comment by aleksandra-tarkowska** [comment]
https://github.com/openmicroscopy/openmicroscopy/commit/49c396df733c7bab374ef0da1111606a4efd303f cherry-picked to https://github.com/openmicroscopy/openmicroscopy/pull/4876

**Pull request merged**